server/src/main/java/org/jetbrains/bsp/bazel/server/sync/BspProjectMapper.kt

```diff
@@ -289,4 +247,34 @@ class BspProjectMapper(
         val items = modules.mapNotNull(scalaLanguagePlugin::toScalaMainClassesItem)
         return ScalaMainClassesResult(items)
     }
+
+    fun rustWorkspace(
+        project: Project,
+        params: RustWorkspaceParams
+    ): RustWorkspaceResult {
+        val rustModules = project.findModulesByLanguage(Language.RUST)
+        val externalModules = project.rustExternalModules.toList()
+        val modules = BspMappings.getModules(project, params.targets)
+            .filter { Language.RUST in it.languages }
+
+        val rustLanguagePlugin = languagePluginsService.rustLanguagePlugin
+        return rustLanguagePlugin.toRustWorkspaceResult(modules, rustModules + externalModules)
+    }
+
+    fun rustToolchain(
```

Comment:
both methods are kinda similar, i believe that it's possible to extract the common logic to a method

Reply:
done